src/github_runner_manager/reactive/runner_manager.py

```diff
@@ -1,140 +1,77 @@
 #  Copyright 2024 Canonical Ltd.
 #  See LICENSE file for licensing details.
 
-"""Module for managing reactive runners."""
+"""Module for reconciling amount of runner and reactive runner processes."""
 import logging
-import os
-import shutil
-import signal
-
-# All commands run by subprocess are secure.
-import subprocess  # nosec
-from pathlib import Path
 
+from github_runner_manager.manager.runner_manager import FlushMode, RunnerManager
+from github_runner_manager.reactive import process_manager
+from github_runner_manager.reactive.consumer import get_queue_size
 from github_runner_manager.reactive.types_ import RunnerConfig
-from github_runner_manager.utilities import secure_run_subprocess
 
 logger = logging.getLogger(__name__)
 
-REACTIVE_RUNNER_LOG_DIR = Path("/var/log/reactive_runner")
 
-PYTHON_BIN = "/usr/bin/python3"
-REACTIVE_RUNNER_SCRIPT_MODULE = "github_runner_manager.reactive.runner"
-REACTIVE_RUNNER_CMD_LINE_PREFIX = f"{PYTHON_BIN} -m {REACTIVE_RUNNER_SCRIPT_MODULE}"
-PID_CMD_COLUMN_WIDTH = len(REACTIVE_RUNNER_CMD_LINE_PREFIX)
-PIDS_COMMAND_LINE = [
-    "ps",
-    "axo",
-    f"cmd:{PID_CMD_COLUMN_WIDTH},pid",
-    "--no-headers",
-    "--sort=-start_time",
-]
-UBUNTU_USER = "ubuntu"
-RUNNER_CONFIG_ENV_VAR = "RUNNER_CONFIG"
+def reconcile(quantity: int, runner_manager: RunnerManager, runner_config: RunnerConfig) -> int:
+    """Reconcile runners reactively.
 
+    The reconciliation attempts to make the following equation true:
```

Comment:
Don't the processes stay alive until the runners finish their job?

Reply:
No, currently the process terminates when the job has started (as per spec). If we were to wait for the job to finish, we would have to check the GH API quite frequently, which could lead to rate limit issues.